packages/web/app/src/components/v2/changelog/changelog.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { ReactElement, useEffect, useState } from 'react';
+import { Button } from '@/components/v2';
+import { useToggle } from '@/lib/hooks';
+import { fetchChangelogsFromGitHub } from './fetch-changelogs-from-github';
+import {
+  Popover,
+  PopoverContent,
+  PopoverFooter,
+  PopoverHeader,
+  PopoverTrigger,
+} from './pop-over-changelog';
+import { useLocalStorage } from './useLocalStorage';
+
+export type Changelog = {
+  title: string;
+  description: string;
+  userReadItem: boolean;
+  href?: string;
+  route?: string;
+  date: string;
+};
+
+export function Changelog(): ReactElement {
+  const [isOpen, toggle] = useToggle();
+  return <ChangelogPopover isOpen={isOpen} onClose={toggle} />;
+}
+
+function ChangelogPopover({ isOpen, onClose }: { isOpen: boolean; onClose: () => void }) {
+  const [changelogItems, setChangelogItems] = useLocalStorage<Changelog[]>('changelogItems', []);
+  const [changeItemDisplay, setChangeItemDisplay] = useState<Changelog | null>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        if (changelogItems?.length) {
+          return;
+        }
+        const data = await fetchChangelogsFromGitHub();
```

Comment:
can we instead embed a json file within the app container from the docs and use that? I don't think GitHub is a good source, as the paths could change and GitHub changes could be newer than the self-hosted version you are running.